### PR TITLE
build: update build and pipeline

### PR DESCRIPTION
**Pull request by bcarter97**
## Description

General housekeeping for the build and pipeline steps of the application in preparation for #48. 

### Scala build

- Update `scalafmt` from 1.16 to 2.4.3
  - Update config
- Add `scalafix`
  - Add `organize-imports` plugin for `scalafix`
- Update `ciBuild` script to use `scalafix`
- Run format/fix on source & test code
- Move dependencies out of `build.sbt` to `Dependencies.scala`
- Remove bintray resolvers

### Pipeline

- Update Travis Scala version
- Update cache script
- Remove unused scala-steward pinned dependencies

### Misc

- Update `.gitignore`
- Update License
- Remove Bintray references from `README`

### CODE REVIEW

**Review by paulaylingdev — APPROVED** [review]
LGTM

File: project/Dependencies.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import sbt._
+
+object Dependencies {
+
+  val KafkaVersion = "2.8.0"
+
+  object Akka {
+    val akka          = "2.6.16"
```
Comment (lacarvalho91) [inline project/Dependencies.scala:8]: minor but `version` seems a more appropriate name?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    val stream        = "com.typesafe.akka" %% "akka-stream"         % akka
+    val streamKafka   = "com.typesafe.akka" %% "akka-stream-kafka"   % "2.1.1"
+    val streamTestkit = "com.typesafe.akka" %% "akka-stream-testkit" % akka % Test
+    val testkit       = "com.typesafe.akka" %% "akka-testkit"        % akka % Test
+    val base          = Seq(stream, streamKafka)
+    val test          = Seq(streamTestkit, testkit)
+  }
+
+  object Cats {
+    val cats   = "2.6.1"
```
Comment (lacarvalho91) [inline project/Dependencies.scala:18]: same as above

File: project/Dependencies.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    val core   = "org.typelevel" %% "cats-core"   % cats
+    val kernal = "org.typelevel" %% "cats-kernel" % cats
+    val all    = Seq(core, kernal)
+  }
+
+  object Refined {
+    val refined    = "0.9.27"
```
Comment (lacarvalho91) [inline project/Dependencies.scala:25]: same as above

**Review by lacarvalho91 — APPROVED** [review]
LGTM, just a minor suggestion.

**Review by Smudgey — APPROVED** [review]

File: project/Dependencies.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    val base       = "eu.timepit" %% "refined"            % refined
+    val pureconfig = "eu.timepit" %% "refined-pureconfig" % refined
+    val all        = Seq(base, pureconfig)
+  }
+
+  val kafkaClients          = "org.apache.kafka"            % "kafka-clients"           % KafkaVersion
+  val scalaLogging          = "com.typesafe.scala-logging" %% "scala-logging"           % "3.9.4"
```
Comment (serragnoli) [inline project/Dependencies.scala:32]: I think it'd be more consistent to list all versions in `object Versions {...}`

File: project/Dependencies.scala (hunk 1 above)
Comment (bcarter97) [reply project/Dependencies.scala]: 👍  renamed to `version` and made private

File: project/Dependencies.scala (hunk 4 above)
Comment (bcarter97) [reply project/Dependencies.scala:30]: I've opted to keep them local to the library object, as I don't think specific dependencies will ever need to know/will be at the same version as others

Comment (serragnoli) [reply project/Dependencies.scala:30]: The new commit containing 
```
private val version = ...
```
looks consistent.

**Review by serragnoli — APPROVED** [review]

**Pull request merged**